app/controllers/principles_controller.rb

```diff
@@ -0,0 +1,30 @@
+class PrinciplesController < ApplicationController
```

Comment:
I think this should be derived from EvaluationsController too and should check for valid group, to avoid unauthorised access. Sorry, I think I've left it out from my review.